packages/camera/camera/example/lib/main.dart

```diff
@@ -636,13 +636,13 @@ class _CameraExampleHomeState extends State<CameraExampleHome>
       ]);
     } on CameraException catch (e) {
       _showCameraException(e);
-    }
+    } finally {
+      if (mounted) {
+        setState(() {});
+      }
 
-    if (mounted) {
-      setState(() {});
+      await previousCameraController?.dispose();
```

Comment:
I couldn't find a reason yet why `CameraController.getMaxZoomLevel` and `CameraController.getMinZoomLevel` throws a `PlatformException` even though this type of exception is always mapped to `CameraException` in these methods.

Added this to always dispose the previous camera controller even if an error occurred while initializing a new one.

Reply:
But maybe it should be added as a separate PR or as a part of https://github.com/flutter/plugins/pull/4240.